### DISCUSSION TITLE
Use Foundry better

### DISCUSSION THREAD

**Discussion by MostPow3rful** [Cyfrin/foundry-full-course-cu#1882] [Ideas]
![photo_2024-06-25_23-05-43](https://github.com/Cyfrin/foundry-full-course-cu/assets/96079030/ff06e0b0-a37e-4eb2-a245-bdff67189d00)

**GM @PatrickAlphaC**
at Foundry101 Course, Section 1, EP 19, you are calling a view/pure function using **cast call**. it's ok but you didn't defined the return value of function and you got the result and decoded it using another command ! it's hard way dude. better to use this :
```bash
cast call $CONTRACT_ADDRESS "retrieve()(uint256)" --rpc-url $RPC_URL
```
**GN**

**Comment by MostPow3rful** [comment]
The Code :
![image](https://github.com/Cyfrin/foundry-full-course-cu/assets/96079030/763649dc-ff7b-4764-a560-801bca1791ba)

Result of Command :
![image](https://github.com/Cyfrin/foundry-full-course-cu/assets/96079030/c7d0cd52-18b0-40d3-8b1b-f9983a942292)

**Reply by EngrPips** [reply]
Interesting, Nice heads-up

**Reply by PatrickAlphaC** [reply]
This is a great shout! Thank you!

**Reply by ciaranightingale** [reply]
wow this is so cool! Learn something new every day - double Thanks!!
